### PR TITLE
Better associations handling

### PR DESCRIPTION
**Pull request by goldoraf**

### CODE REVIEW

File: packages/cozy-client/src/associations/HasManyAssociation.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+export default class HasManyAssociation {
+  constructor(target, name, { get, query, mutate }) {
+    this.target = target
+    this.name = name
+    this.get = get
+    this.query = query
+    this.mutate = mutate
+  }
+
+  get data() {
```
Comment (ptbrowne) [inline packages/cozy-client/src/associations/HasManyAssociation.js]: With getter it seems it is not possible to target IE8.

File: packages/cozy-client/src/associations/HasManyAssociation.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      ...target,
+      relationships: {
+        ...target.relationships,
+        [this.name]: {
+          ...target.relationships[this.name],
+          ...updateFn(target.relationships[this.name])
+        }
+      }
+    }
+  }
```
Comment (ptbrowne) [inline packages/cozy-client/src/associations/HasManyAssociation.js]: Shouldn't this two functions be located in Association, they seem generic enough ?

File: packages/cozy-client/src/associations/HasManyAssociation.js (hunk 1 above)
Comment (goldoraf) [reply packages/cozy-client/src/associations/HasManyAssociation.js]: And I don't give a shit ;)

File: packages/cozy-client/src/associations/HasManyAssociation.js (hunk 2 above)
Comment (goldoraf) [reply packages/cozy-client/src/associations/HasManyAssociation.js]: They will probably move in Association, yes.

**Pull request merged**